File: order-service/frontend/src/domain/restaurant.ts
import type {Address} from "./address.ts";

export interface Restaurant {
    id: string;
    name: string;
    address: Address;
    cuisineType: string;
    priceIndicator: string;
    openingHours: OpeningHours;
    isOpen: boolean;
}

export interface OpeningHours {
    days: {
        day: string;
        openingPeriods: {
          openTime: string
          closeTime: string
        }[]
    }[]
}

File: backend/src/main/java/be/kdg/sa/backend/domain/restaurant/Restaurant.java
package be.kdg.sa.backend.domain.restaurant;

import org.jmolecules.ddd.annotation.ValueObject;

@ValueObject
public record Restaurant(
        RestaurantId id,
        String name,
        Address address,
        String cuisineType,
        String priceIndicator,
        OpeningHours openingHours,
        boolean isOpen
) {
}
